Alloy/MTLFunctionConstantValues+Extensions.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  MTLFunctionConstantValues+Extensions.swift
+//  Alloy
+//
+//  Created by Eugene Bokhan on 08/09/2019.
+//
+
+import Metal
+
+public extension MTLFunctionConstantValues {
+
+    public func setConstantValue<T>(_ value: T,
```

Comment:
it doesn't feel swifty + easy to miss with original API, maybe just `set(value:)`? just like we have at encoder

+ I would introduce specialized methods like set(value: Float) where user doesn't need to pass MTLDataType